File: src/main/java/ru/romanow/serialization/services/JsonSerializer.java
package ru.romanow.serialization.services;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

/**
 * Created by ronin on 09.09.16
 */
public class JsonSerializer {

    private static final Gson gson =
            new GsonBuilder()
                    .setPrettyPrinting()
                    .create();

    public static String toJson(Object object) {
        return gson.toJson(object);
    }

    public static <T> T fromJson(String json, Class<T> cls) {
        return gson.fromJson(json, cls);
    }
}

File: src/main/java/ru/romanow/serialization/services/JsonSerializer.kt
package ru.romanow.serialization.services

import com.fasterxml.jackson.databind.SerializationFeature.INDENT_OUTPUT
import com.fasterxml.jackson.module.kotlin.jacksonMapperBuilder
import com.jayway.jsonpath.JsonPath

private val mapper = jacksonMapperBuilder()
    .configure(INDENT_OUTPUT, true)
    .build()

fun toJson(data: Any): String = mapper.writeValueAsString(data)

fun <T> fromJson(json: String, cls: Class<T>): T = mapper.readValue(json, cls)

fun jsonPath(json: String, path: String): List<String> = JsonPath.read<List<String>>(json, path)